core/templates/dev/head/pages/exploration_editor/editor_tab/UnresolvedAnswersOverviewDirective.js

```diff
@@ -0,0 +1,90 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the state graph visualization.
+ */
+
+oppia.directive('unresolvedAnswersOverview', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/exploration_editor/editor_tab/' +
+        'unresolved_answers_overview_directive.html'),
+      controller: [
+        '$scope', 'EditorStateService', 'ExplorationStatesService',
+        'StateRulesStatsService',
+        function(
+            $scope, EditorStateService, ExplorationStatesService,
+            StateRulesStatsService) {
+          var MAXIMUM_UNRESOLVED_ANSWERS = 5;
+          var MINIMUM_UNRESOLVED_ANSWER_FREQUENCY = 2;
+
+          $scope.unresolvedAnswersData = null;
+          $scope.latestRefreshDate = new Date();
+
+          $scope.computeUnresolvedAnswers = function() {
+            var state = ExplorationStatesService.getState(
+              EditorStateService.getActiveStateName()
+            );
+
+            if (!StateRulesStatsService.stateSupportsIssuesOverview(state)) {
+              $scope.unresolvedAnswersData = [];
+            } else {
+              StateRulesStatsService.computeStateRulesStats(
+                state
+              ).then(function(stats) {
```

Comment:
Optional, but would be better to create a frontend domain object immediately once the backend dict is returned, and use that throughout instead.

Reply:
I'll leave that as future work, since there's still things I believe would fit better in visualizations domain object.